css/badge.css

```diff
@@ -0,0 +1,76 @@
+﻿.badge-element{
+    position: relative;
+    
+}
+
+.badge-icon {    
+    background-color: var(--info-text);
+    border-radius: 50rem;
+    color: var(--white-color);
+    display: inline-block;
+    font-size: 0.75rem;
+    font-weight: 600;
+    padding: 0.35em 0.5em;
+    text-align: center;
+    line-height: 1;
+    position: absolute;
+    left: 70%;
+    top: 0;
+    transform: translate(-50%, -50%);
+    white-space: nowrap;
+}
+
+.badge-red{
+    background-color: red;
+}
+
+.badge-green {
+    background-color: #3DC94F;
+}
+
+.badge-green-avatar {
+    background-color: #3DC94F;
+}
+.badge-red-avatar{
+    background-color: red;
+}
+.badge-yellow-avatar {
+    background-color: yellow;
+}
+.badge-grey-avatar {
+    background-color: grey;
+}
+.badge_description {
+    display: none;
+}
+
+.badge-green-avatar:hover + .badge_description,
+.badge-red-avatar:hover + .badge_description,
+.badge-yellow-avatar:hover + .badge_description,
+.badge-grey-avatar:hover + .badge_description {
+    display: block;
+    background-color: black;
+    color: white;
+    position: absolute;
+    left: 72%;
+    top: 50%;
+}
```

Comment:
If these are common properties for all the classes, You can use a common class for all these badges.